insertion_sort.go

```diff
@@ -11,23 +9,56 @@ type LessFunc func(a, b int) bool
 
 // InsertionSort rearranges the list to become sorted.
 func InsertionSort(l *list.List, less LessFunc) {
```

Comment:
This is too complicated. Furthermore, you don't need Remove here. Try to create and maintain a new sorted list instead of modifying an existing one. Can you use SortedInsert?

Reply:
Wasn't it required to implement soring in-place?

Reply:
Yes, it was an optional task. However, your solution is not in-place. `l.PushFront` and `l.InsertAfter` create new nodes.

I propose to do the following:
1) simplify `InsertionSort` so it uses `SortedInsert` (the implementation still allocates new nodes). You will get the full score for the mandatory part then.
2) implement an in-place version by adding new methods to `List`: `MoveFront` and `MoveAfter`, and using them from `InsertionSort`.